core/server.go

```diff
@@ -42,24 +41,30 @@ type (
 	ConnMiddleware func(ConnHandler) ConnHandler
 )
 
+// VersionNegotiateFunc is the version negotiate function.
+// Use w to respond to client, and return the error in version negotiate process for ending handshake.
+// NOTICE: do not respond HandshakeAckFrame to client in this function, just return nil if version negotiation is ok.
+type VersionNegotiateFunc func(w frame.Writer, cVersion string, sVersion string) error
```

Comment:
```suggestion
type VersionNegotiateFunc func(cVersion string, sVersion string) error
```
Is it better for the server to handle the error？

Reply:
The `VersionNegotiateFunc` is a public type, user should implement their own `VersionNegotiateFunc` and pass it in `server` by calling `server.ConfigVersionNegotiateFunc()`, The `w` is used to respond to client.